scio-bigtable/src/main/scala/com/spotify/scio/bigtable/TableAdmin.scala

```diff
@@ -154,20 +166,24 @@ object TableAdmin {
         case (table, columnFamilies) =>
           val tablePath = s"$instancePath/tables/$table"
 
-          if (!existingTables.contains(tablePath)) {
-            log.info("Creating table {}", table)
-            client.createTable(
-              CreateTableRequest
-                .newBuilder()
-                .setParent(instancePath)
-                .setTableId(table)
-                .build()
-            )
-          } else {
-            log.info("Table {} exists", table)
+          val exists = existingTables.contains(tablePath)
+          createDisposition match {
+            case _ if exists =>
+              log.info("Table {} exists", table)
+            case CreateDisposition.CreateIfNeeded =>
+              log.info("Creating table {}", table)
+              client.createTable(
+                CreateTableRequest
+                  .newBuilder()
+                  .setParent(instancePath)
+                  .setTableId(table)
+                  .build()
+              )
+            case CreateDisposition.Never =>
+              log.info("Table {} does not exist", table)
           }
 
-          ensureColumnFamilies(client, tablePath, columnFamilies)
+          ensureColumnFamilies(client, tablePath, columnFamilies, createDisposition)
```

Comment:
We don't need to pass `createDisposition` here right? Can just fail fast on line 183 and throw an exception? Also it's probably a desired behavior to throw exception than do nothing silently?

Reply:
this is allowing the case where table exists and users want to modify CF's... they will only be able to modify it if they pass `CreateIfNeeded`